runtime-modules/membership/src/lib.rs

```diff
@@ -741,6 +784,153 @@ decl_module! {
             Self::deposit_event(RawEvent::MemberInvited(invited_member_id, params));
         }
 
+        /// Gift a membership using own funds. Gifter does not need to be a member.
+        /// Can optinally apply a lock on a portion of the funds transferred to root and controller
+        /// accounts. Gifter also pays the membership fee.
+        #[weight = WeightInfoMembership::<T>::gift_membership(
+            Module::<T>::text_length_unwrap_or_default(&params.handle),
+            params.metadata.len().saturated_into(),
+        )]
+        pub fn gift_membership(
+            origin,
+            params: GiftMembershipParameters<T::AccountId, BalanceOf<T>>,
+        ) {
+            let gifter = ensure_signed(origin)?;
+
+            let handle_hash = Self::get_handle_hash(
+                &params.handle,
+            )?;
+
+            // Check that gifter has sufficient funds
+            let membership_fee = Self::membership_price();
+            let gifter_free_balance = balances::Pallet::<T>::usable_balance(&gifter);
+            let total_credit = params
+                .credit_controller_account
+                .saturating_add(params.credit_root_account)
+                .saturating_add(membership_fee);
+
+            ensure!(
```

Comment:
And also should we allow the `gifter` to be dropped in case the balance gets below the existential deposit limit?

Reply:
Excellent catch here with the proper checks.
For allowing death, maybe I will re-consider. Perhaps it can be a bool argument to the dispatch call?

Reply:
I think it will be better to make sure the `gifter` has enough cash to cover everything including ex. dep.
So that the `gift_membership` extr. can only used in one single (simple) way. But besides that I don't have any other 
counter arguments